[Wang Wen Kai Daniel] iP

dwangwk opened the pull request:
# Tony
> "The inner machinations of my mind are an enigma." – Patrick Star
## The Sickest Chat Bot you've ever seen :heart_eyes:

Tony allows you to sit back, relax and enjoy life. No more need to remember your tasks,
let Tony do it for you and actually start living. He's:
* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/dwangwk/ip/releases/tag/A-Jar/).
1. double-click it.
1. add your tasks.
1. let him manage your tasks for you

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Tony.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

danielwangcinch committed:
test edit

danielwangcinch committed:
feat: greeting/goodbye

danielwangcinch committed:
feat: echo

danielwangcinch committed:
feat: add ,list

danielwangcinch committed:
feat: mark as done

danielwangcinch committed:
feat: mark as done w/ inheritance

danielwangcinch committed:
feat: exceptions and handling

danielwangcinch committed:
feat: delete

dwangwk committed:
Merge pull request #1 from dwangwk/week-2

Week 2

danielwangcinch committed:
feat: A-TextUiTesting

danielwangcinch committed:
feat: A-Enums

danielwangcinch committed:
fix: Tasktype Enum

dwangwk committed:
Merge pull request #2 from dwangwk/week-2

Week 2

danielwangcinch committed:
feat: Level-7 Save

danielwangcinch committed:
Part of the instructions
Merge branch 'branch-Level-7'

danielwangcinch committed:
feat: Level 8. Dates and Times

danielwangcinch committed:
Merge branch 'branch-Level-8'

danielwangcinch committed:
feat: A-MoreOOP

danielwangcinch committed:
feat: A-Packages

dwangwk committed:
Merge pull request #3 from dwangwk/add-gradle-support

Add gradle support

danielwangcinch committed:
feat: A-Gradle

danielwangcinch committed:
feat: A-JUnit

danielwangcinch committed:
feat: A-JavaDoc

danielwangcinch committed:
feat: A-CodingStandard

danielwangcinch committed:
feat: Level-9

danielwangcinch committed:
Merge remote-tracking branch 'origin/branch-Level-9'

danielwangcinch committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

danielwangcinch committed:
fix: merge conflicts

danielwangcinch committed:
fix: checkstyle

wallacexuhanxiao commented on src/main/java/tony/Parser.java:
```diff
@@ -0,0 +1,71 @@
+            int hour = Integer.parseInt(timeParts[0]);
+            int minutes = Integer.parseInt(timeParts[1]);
+            return LocalDateTime.of(year, month, day, hour, minutes);
+        } catch (Exception e) {
```
Maybe can catch specific ```Exception```? 
Isn't ```catch (Exception e)``` bug prone as it may neglect other types of ```Exception```s that potentially happen?

wallacexuhanxiao commented on src/main/java/tony/Parser.java:
```diff
@@ -0,0 +1,71 @@
+            throw new BadDateException();
+        }
+    }
+}
```
Great ```Parser``` I like it 👍 .

wallacexuhanxiao commented on src/main/java/tony/Storage.java:
```diff
@@ -0,0 +1,58 @@
+        }
+    }
+
+}
```
Great ```storage``` I like it 👍 .

wallacexuhanxiao commented on src/main/java/tony/TaskFactory.java:
```diff
@@ -0,0 +1,53 @@
+     * @return A task object of the specified type.
+     * @throws IllegalArgumentException If an invalid task type is provided.
+     */
+    public Task createTask(TaskType type, String... args) throws InvalidTaskException {
```
Nice utilization of ```varargs```

wallacexuhanxiao commented on src/main/java/tony/TodoList.java:
```diff
@@ -0,0 +1,197 @@
+ * Represents a list of tasks in the Tony application.
+ */
+public class TodoList {
+    private List<Task> list = new ArrayList<>();
```
Maybe the list could be named plural like ```tasks```?

wallacexuhanxiao commented on src/main/java/tony/TodoList.java:
```diff
@@ -0,0 +1,197 @@
+        return null;
+    }
+
+    private static void line() {
```
Isn't the function name should be a verb? 
Maybe ```printLine```should be better. 🤔

wallacexuhanxiao commented on src/main/java/tony/tasks/Task.java:
```diff
@@ -0,0 +1,79 @@
+        return description;
+    }
+}
+
```
Maybe delete these empty lines?

wallacexuhanxiao reviewed (commented):
LGTM. 👍

dwangwk committed:
feat: Update README.md

danielwangcinch committed:
feat: Readme, Checkstyle tag

modembcc commented on src/main/java/tony/Ui.java:
```diff
@@ -0,0 +1,35 @@
+                + "_________________________\n";
+        System.out.println(greeting);
+    }
+    /**
```
Should a blank line be added to separate methods (for consistency like in TodoList.java)?
```suggestion
    /**

```

modembcc commented on src/main/java/tony/TodoList.java:
```diff
@@ -0,0 +1,197 @@
+            }
+        }
+        return null;
+    }
```
I like the if-else indentations

modembcc commented on src/main/java/tony/Tony.java:
```diff
@@ -0,0 +1,103 @@
+                    break;
+                default:
+                    throw new IllegalArgumentException("Invalid command: " + command);
+                }
```
I like the switch-case indentations!

modembcc commented on src/main/java/tony/Ui.java:
```diff
@@ -0,0 +1,35 @@
+     * Displays a greeting message when the program starts.
+     */
+    public static void greeting() {
+        String greeting = "_______________________\n"
+                + "what is up dawg! I'm tony.Tony!\n"
+                + "What can I do for you mate?\n"
+                + "_________________________\n";
+        System.out.println(greeting);
+    }
```
Should line() method be used for consistency?
```suggestion
        String greeting = "What is up dawg! I'm tony. Tony!\n"
                + "What can I do for you mate?\n";
        line();
        System.out.println(greeting);
        line();
    }
```

modembcc reviewed (commented):
I like that you are following the coding standards. Good job!

danielwangcinch committed:
feat: JavaFX, GUI

danielwangcinch committed:
feat: GFMD

danielwangcinch committed:
feat: Assertions

danielwangcinch committed:
refactor: Improve code quality by restructuring methods and enhancing error handling

In this commit, several improvements have been made to enhance the overall code quality:
- Encapsulated command handling logic into separate methods to improve readability and maintainability.
- Introduced enums for commands (`TaskType`) to enhance clarity and type safety.
- Implemented more informative error handling, providing specific error messages for different scenarios.
- Reduced code duplication by refactoring task creation logic into dedicated methods.
- Ensured consistency in error handling throughout the `run` method.

These changes aim to enhance the maintainability, readability, and robustness of the codebase.

danielwangcinch committed:
refactor: Utilize Java streams for improved code readability and conciseness

In this commit, the following changes have been made to the TodoList class:
- Replaced traditional loops with Java streams in print(), printTasksToString(), and find() methods to enhance code readability and conciseness.
- Used stream operations such as map(), filter(), and collect() to transform, filter, and join task data efficiently.
- Encapsulated task iteration and string construction logic within stream operations, resulting in cleaner and more expressive code.

These stream-based changes contribute to better code maintainability and align with modern Java programming practices.

dwangwk committed:
Merge pull request #4 from dwangwk/branch-A-CodeQuality

refactor: Improve code quality by restructuring methods and enhancing…

dwangwk committed:
Merge pull request #6 from dwangwk/branch-A-Assertions

feat: Assertions

dwangwk committed:
Merge branch 'master' into branch-A-Streams

dwangwk committed:
Merge pull request #5 from dwangwk/branch-A-Streams

refactor: Utilize Java streams for improved code readability and conc…

dwangwk committed:
Merge branch 'master' into branch-A-Checkstyle

dwangwk committed:
Merge pull request #7 from dwangwk/branch-A-Checkstyle

feat: Readme, Checkstyle tag

danielwangcinch committed:
feat: update

danielwangcinch committed:
fix: javadocs

danielwangcinch committed:
feat: Personality

danielwangcinch committed:
feat: Readme User Guide

danielwangcinch committed:
feat: Exit on Bye

danielwangcinch committed:
feat: newest tony.jar

danielwangcinch committed:
fix: dialogbox cut off, task errors
